src/sql-parser/src/lexer.rs

```diff
@@ -191,16 +170,10 @@ fn lex_ident(buf: &mut LexBuf) -> Token {
     let word = buf.take_while(
         |ch| matches!(ch, 'A'..='Z' | 'a'..='z' | '\u{80}'..='\u{ff}' | '0'..='9' | '$' | '_'),
     );
-    let word_uppercase = word.to_uppercase();
-    Token::Word(Word {
-        value: word.into(),
-        quoted: false,
-        keyword: if ALL_KEYWORDS.contains(&word_uppercase.as_str()) {
-            word_uppercase
-        } else {
-            "".into()
-        },
-    })
+    match word.parse() {
+        Ok(kw) => Token::Keyword(kw),
+        Err(_) => Token::Ident(word.to_lowercase()),
```

Comment:
Nope; that was just unnecessarily complexity in the old code. We do keep track, in a sense, because we only lowercase unquoted idents; the code to print an ident is already smart enough to check for uppercase/special chars and quote if necessary.

Reply:
To clear up this confusion (I had it too): in this function (`lex_ident`) we already know that the thing being parsed isn't quoted because we have encountered a char that indicates the start of a keyword/ident (and not a quote). The pg lex docs say that identifiers (which is anything that isn't a predefined keyword) are case insensitive and convention is to lowercase them. We do have to care about case when lexing a quoted ident which happens in `lex_quoted_ident`, which returns Token::Ident but without the lowercasing. So we lose the exact original casing for unquoted idents when, say, printing a statement back out, but that's fine and makes parsing/semantic checking easier since the case insensitivity has already been handled by the lexer.